addons/escoria-core/game/core-scripts/esc/commands/set_direction.gd

```diff
@@ -0,0 +1,75 @@
+# `set_direction object direction_id [wait]`
+#
+# Turns a movable `ESCItem` or `ESCPlayer` to face a given target direction id
+# (between 0 and 3 for a 4-directional character, or between 0 and 7 for an
+# 8-directional character.
+#
+# **Parameters**
+#
+# - *object*: Global ID of the object to turn
+# - *target_direction_id*: id of the direction as defined in the player animations
+# - *wait*: Number of seconds to wait for while playing each animation occurring
+#   between the current angle of `object` and the target angle. A value of
+#   `0` will complete the turn immediately (default: `0`)
+#
+# @ESC
+extends ESCBaseCommand
+class_name SetDirectionCommand
+
+
+# Return the descriptor of the arguments of this command
+func configure() -> ESCCommandArgumentDescriptor:
+	return ESCCommandArgumentDescriptor.new(
+		2,
+		[TYPE_STRING,  [TYPE_INT],  [TYPE_REAL, TYPE_INT]],
+		[null, null, 0.0]
+	)
+
+
+# Validate whether the given arguments match the command descriptor
+func validate(arguments: Array):
+	if not .validate(arguments):
+		return false
+
+	if not escoria.object_manager.has(arguments[0]):
+		escoria.logger.error(
+			self,
+			"[%s]: invalid object. Object with global id %s not found."
+					% [get_command_name(), arguments[0]]
+		)
+		return false
+	elif not escoria.object_manager.get_object(arguments[0]).node.is_movable:
+		escoria.logger.error(
+			self,
+			"[%s]: invalid object. Object with global id %s is not movable."
+					% [get_command_name(), arguments[0]]
+		)
+		return false
+	elif arguments[1] < 0 or arguments[1] > escoria.object_manager.get_object(arguments[0]).node \
+			.get_directions_quantity() - 1:
+		escoria.logger.error(
+			self,
+			"[%s]: invalid direction id for object %s. Valid values are integer values between 0 and %s."
+					% [get_command_name(), arguments[0], escoria.object_manager.get_object(arguments[0]) \
+						.node.get_directions_quantity() - 1]
+		)
+		return false
+	
+	return true
+
+
+# Run the command
+func run(command_params: Array) -> int:
+	# HACK Countering the fact that angle_to_point() function gives
+	# angle against X axis not Y, we need to check direction using (angle-90°).
+	# Since the ESC command already gives the right angle, we add 90.
```

Comment:
Where does this hack come into play? I don't see where 90 is added, unless it's in another location?

Reply:
This was copy-pasted from `set_angle` script. Good catch, not needed, removing. Thanks!